spec/spec_helper.rb

```diff
@@ -1,27 +1,56 @@
+
+require_relative "../lib/autosign"
 require 'coveralls'
 Coveralls.wear!
-# This file was generated by the `rspec --init` command. Conventionally, all
-# specs live under a `spec` directory, which RSpec adds to the `$LOAD_PATH`.
-# The generated `.rspec` file contains `--require spec_helper` which will cause
-# this file to always be loaded, without a need to explicitly require it in any
-# files.
-#
-# Given that it is always loaded, you are encouraged to keep this file as
-# light-weight as possible. Requiring heavyweight dependencies from this file
-# will add to the boot time of your test suite on EVERY test run, even for an
-# individual file that may not need all of that loaded. Instead, consider making
-# a separate helper file that requires the additional dependencies and performs
-# the additional setup, and require it from the spec files that actually need
-# it.
-#
-# The `.rspec` file also contains a few flags that are not defaults but that
-# users commonly want.
-#
-# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
-#
-require_relative "../lib/autosign"
 @fixture_path = File.expand_path(File.join(__FILE__, '..', 'fixtures'))
 
+# Generate X509 CSR (certificate signing request) with SAN (Subject Alternative Name) extension and sign it with the RSA key
+# common_name, organization, country, state_name, locality, _domain_list = []
+# Usage example
+# generate_csr({
+#   common_mame: 'acme.com',
+#   organization: 'ACME Corp.',
+#   country: 'US',
+#   state_name: 'California',
+#   locality: 'San Francisco',
+#   csr_attributes: {'challengePassword' => 'sdfasdfds'}
+# }
+# )
+def generate_csr(opts)
```

Comment:
This helper code generates a real csr on the fly and allows us to pass in csr attributes.   Very helpful when we need a passing test and also want a good csr.